Puzzle/src/org/fundacionjala/devops/bootcamp/Move/Movement.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+
+    //Method to check if the movement is valid:
+    public boolean isValidMovement(String position, String direction,int size) {
+        String[] positionArray = position.split(",");
+        int row = Integer.parseInt(positionArray[0]);
+        int column = Integer.parseInt(positionArray[1]);
+        if (direction.equals("N")){
+            if(row == 0) {
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                System.out.println("You can't go up!");
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                return false;
+            }
+        }else if(direction.equals("S")){
+            if(row == size-1) {
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                System.out.println("You can't go down!");
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                return false;
+            }
+        }else if(direction.equals("E")){
+            if(column == size-1) {
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                System.out.println("You can't go right!");
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                return false;
+            }
+        }else if(direction.equals("O")){
+            if(column == 0) {
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                System.out.println("You can't go left!");
+                System.out.println("XXXXXXXXXXXXXXXXXXX");
+                return false;
+            }
+        }
+        return true;
+    }
+    //Method that receive the position of zero in the board, and return a String with the possible movements:
+    public String posibleMovements(int[][] board) {
+        String movements = "You can go to: ";
+        Position newPosition= new Position();
+        String zeroPosition= String.valueOf(newPosition.getHolePosition());
+        int boardSize = board.length;
+        for (int i = 0; i < board.length; i++) {
+            for (int j = 0; j < board[i].length; j++) {
+                if (board[i][j] == 0) {
+                    System.out.println("The position of the zero is: " + (i+1) + "," + (j+1));
+                    if (i == 0) {
+                        movements += "S ";
+                    }else if (i == boardSize - 1) {
+                        movements += "N ";
+                    }else{
+                        movements += "N, S ";
+                    }
+                    if (j == 0) {
+                        movements += ", E ";
+                    }else if (j == boardSize - 1) {
+                        movements += ", O ";
+                    }else{
+                        movements += ", O, E ";
+                    }
+                }
+            }
+        }
+        System.out.println(movements);
+        return movements;
+    }
+    //Method to move the zero, that receive the board and the movement("N", "S", "E" or "O"):
+    public int moveZero(int[][] board, String movement) {
```

Comment:
It seems that this method moves the position of the hole, you probably we can change the method's name: "changeHolePosition" ??

Reply:
Done